core/src/style/textStyle.cpp

```diff
@@ -111,6 +111,25 @@ auto TextStyle::applyRule(const DrawRule& _rule, const Properties& _props) const
         }
     }
 
+    if (!_rule.get(StyleParamKey::repeat_group, p.labelOptions.repeatGroup)) {
+        // TODO: default to 'draw.key'
+        // TODO: Optimize - this is kind of heavy on allocations
+        for (auto* name : _rule.getLayerNames()) {
+            p.labelOptions.repeatGroup += name;
+            p.labelOptions.repeatGroup += "/";
+        }
+        //LOG("rg: %s", p.labelOptions.repeatGroup.c_str());
+    }
```

Comment:
The used hash value would be the same. I mean we currently keep the whole repeatGroup string in labelOptions, calculate the hash for the string in checkRepeatGroup and then only match groups based on the hash value there. This could be considered a bug or an optimization (by only comparaing the hash value instead of the repeatGroup strings) 

My suggestion is to only store the repeatGroup hash value in labelOptions